visualization/app/codeCharta/e2e/url.e2e.ts

```diff
@@ -52,13 +55,13 @@ describe("codecharta", () => {
 				request.respond({
 					contentType: "application/json",
 					headers: { "Access-Control-Allow-Origin": "*" },
-					body: JSON.stringify(require("../assets/sample2.cc.json"))
+					body: JSON.stringify(sample1)
```

Comment:
That is true.
Just for my clarification: When importing a .json file, it's getting parsed.
When opening it with fs it's still stringified.

So in that case here we'd so something like `body: fs.readFileSync("../assets/sample2.cc.json2)`

Reply:
`fs` reads the data into a buffer. That is turned into a string if you pass through the `encoding` option and set it to `utf8` (`fs.readFileSync("../assets/sample2.cc.json2", "utf8")`). You might want to use the promise based function instead though. Using the sync versions is almost always not ideal.

Reply:
```
let sample1: string
fs.readFile("../assets/sample1.cc.json", "utf8", (err, data) => {
	sample1 = data
})
```

This seems a bit long.

Are you fine with wrapping that in a `jsonReader.ts` which exports this function:
```
export function readJSON(path: string): string {
	let tmp: string

	fs.readFile(path, "utf8", (err, data) => {
		tmp = data
	})
		
	return tmp
}
```

This would give us way shorter calls such as
```
const sample1 = readJSON("../assets/sample1.cc.json")
```

Reply:
Your example code would not work. It would always return `undefined`. It's an async call but the function is executed synchronously. Using the promise based version is much nicer:

```js
page.on("request", async (request) => {
	if (request.url().includes("/fileOne.json")) {
		const body = await fs.promises.readFile("../assets/sample2.cc.json2", "utf8")
		request.respond({
			contentType: "application/json",
			headers: { "Access-Control-Allow-Origin": "*" },
			body
			...
```

Reply:
Ich habe versucht es umzusetzen. Es sind wohl einige TS-Entwickler unzufrieden, weil es keinen ordentlichen import dafür gibt. 
Jedenfalls funktioniert `import { promises as fs } from "fs"` für mich nicht.

https://github.com/nodejs/node/issues/21014#issuecomment-529012253

Reply:
@NearW it's possible to import it with `import fsPromise from "fs/promises"`

See https://nodejs.org/api/fs.html#fs_fs_promises_api

Reply:
https://github.com/nodejs/node/pull/31553

Das geht wohl leider erst ab Node 14

Reply:
Ha, I see. In that case it's possible to use:

```js
import fs from "fs";

fs.promises.readFile()
```